package/libgluonutil/src/libgluonutil.c

```diff
@@ -212,6 +214,36 @@ char * gluonutil_get_domain(void) {
 	return ret;
 }
 
+char * gluonutil_get_primary_domain(void) {
+	if (!gluonutil_has_domains())
+		return NULL;
+
+	char *domain_code = gluonutil_get_domain();
+	if (!domain_code)
+		return NULL;
+
+	const char *domain_path_fmt = "/lib/gluon/domains/%s.json";
+	char domain_path[strlen(domain_path_fmt) + strlen(domain_code)];
+	snprintf(domain_path, sizeof(domain_path), domain_path_fmt, domain_code);
+	free(domain_code);
+
+	char primary_domain_path[PATH_MAX] = "";
```

Comment:
Let's make this buffer `PATH_MAX+1`, but still pass `PATH_MAX` to `readlink`, so it is 0-terminated even for oversized paths (however that is supposed to happen). It is not necessary to initialize the whole array: Setting byte `len` for `len` returned by readlink is sufficient.